jerry-core/vm/vm.c

```diff
@@ -2953,6 +2953,7 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           }
 
           ecma_object_t *catch_env_p = ecma_create_decl_lex_env (frame_ctx_p->lex_env_p);
+          catch_env_p->type_flags_refs = (uint16_t) (catch_env_p->type_flags_refs | ECMA_OBJECT_FLAG_NON_CLOSURE);
```

Comment:
Two questions:
1. Is this code debugger-specific? If so, it should be wrapped into `#ifdef JERRY_DEBUGGER` / `#endif /* JERRY_DEBUGGER */`.
2. Isn't this equivalent to `catch_env_p->type_flags_refs = (uint16_t) ECMA_OBJECT_FLAG_NON_CLOSURE;`?